src/components/date-picker/dropdown.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import {getMonthNames, getMonthFromOffset, getDayNames, getDaysInMonth, getNumOfPreviousDays, getNumOfFollowingDays} from '../../common/date-helpers';
+import {observable, action, computed} from 'mobx';
+import {observer} from 'mobx-react';
+import {Day} from './day';
+const styles = require('./date-picker.st.css').default;
+
+export interface DropdownProps {
+    value: Date;
+    onChange(date: Date): void;
+}
+
+const monthNames = getMonthNames();
+
+@observer
+export class Dropdown extends React.Component<DropdownProps, {}>{
+    @observable date: Date = this.props.value;
+
+    @action
+    setDate (date: Date) {
+        this.date = date;
+    }
+
+    @action
+    setDay = (day: number) => {
+        const date = new Date(this.date.getFullYear(), this.date.getMonth(), day);
+        this.props.onChange(date);
+    };
```

Comment:
`setDay` is bad name because it has side effect. 
I would use `onChange` which calls `this.props.onChange` and `setDate`